py/selenium/webdriver/support/expected_conditions.py

```diff
@@ -102,7 +102,7 @@ def __init__(self, locator):
     def __call__(self, driver):
         return _find_elements(driver, self.locator)
         
-class visibility_of_all_elements_located(object):
+class visibility_of_any_elements_located(object):
```

Comment:
This is clearly not the author's intent and returning elements instead of a boolean is not out of line with the rest of the expected conditions.

See presence_of_element_located: https://github.com/Mobrockers/selenium/blob/f878b53d9dd66ef25444b82e50218acd36f6bc4e/py/selenium/webdriver/support/expected_conditions.py#L49
visibility_of_element_located:https://github.com/Mobrockers/selenium/blob/f878b53d9dd66ef25444b82e50218acd36f6bc4e/py/selenium/webdriver/support/expected_conditions.py#L61

etcetc

Or do you mean as a final condition if no visible elements van be found by the given locator?
In that case perhaps you're right yes.